liltab/data/datasets.py

```diff
@@ -66,8 +67,10 @@ def __init__(
 
         self.y = self.df[self.target_columns]
         if self.encode_categorical_target:
-            self.y = pd.get_dummies(self.y.astype("category"))
-        self.y = torch.from_numpy(self.y.to_numpy()).type(torch.float32)
+            self.one_hot_encoder = OneHotEncoder(sparse=False).set_output(transform="pandas")
```

Comment:
RandomFeaturesPandasDataset needs to have the same logic applied (target_column and cases, self.raw_y etc.)